software/controller/test_libs/mocks/mock_stepper.cpp

```diff
@@ -0,0 +1,370 @@
+/* Copyright 2022, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "mock_stepper.h"
+
+#include <cfloat>  // for FLT_EPSILON
+#include <cmath>   // for fmod and fabsf
+
+namespace StepMotor {
+
+// Keep in sync with stepper.cpp in hal library
+static constexpr size_t param_len[32] = {
```

Comment:
If these 2 data structures are closely related to the implementation, maybe we ought to move them somewhere there as well? My suggestion would be to maybe move the `stepper.h` implementations out into the `actuators` directory, or even a separate directory and maybe have some documentation around linking to manufacturer data sheets there?
Soon enough we should have different directories for `abstract_hal` and `stm32_hal`, so we can sort of move this piece in that direction already?